even_fib_nums.js

```diff
@@ -5,31 +5,91 @@
  * @param  {Number} maxFibValue
  * @return {Number} sum
  */
-function _sumFibs( maxFibValue ) {
+function _sumFibs(maxFibValue) {
   var sum = 0;
+  var fibArray = [1, 2];
+  function fibMaker() {
+    var fib1 = 1;
+    var fib2 = 2;
 
-  // do your work here
+    var fibonacci = fib1 + fib2;
+    while (fibonacci < maxFibValue) {
+      fibonacci = fib1 + fib2;
+      fib1 = fib2;
+      fib2 = fibonacci;
+      fibArray.push(fibonacci);
+    }
+    return fibArray;
+  }
+  fibMaker();
+
+  for (i = 0; i < fibArray.length; i++) {
+    if (fibArray[i] % 2 === 0) {
+      sum += fibArray[i];
+    }
+  }
 
   return sum;
 }
 
 // bonus round
-function _highestFibonacciNumber (maxFibValue){
+function _highestFibonacciNumber(maxFibValue) {
   var highest = 0;
 
   //define your base case, validate your input
 
+  var fibArray = [1, 2];
+  function fibMaker() {
+    var fib1 = 1;
+    var fib2 = 2;
 
-  //do your work here
+    var fibonacci = fib1 + fib2;
+    while (fibonacci <= maxFibValue) {
+      fibonacci = fib1 + fib2;
+      fib1 = fib2;
+      fib2 = fibonacci;
+      if (fibonacci < maxFibValue) {
+        fibArray.push(fibonacci);
+      }
+    }
+    return fibArray;
+  }
+  fibMaker();
+  highest = fibArray.pop();
 
   return highest;
-};
+}
 
+function _oddFibs(maxFibValue) {
+  var sum = 0;
+  var fibArray = [1, 2];
+  function fibMaker() {
+    var fib1 = 1;
+    var fib2 = 2;
+
+    var fibonacci = fib1 + fib2;
+    while (fibonacci <= maxFibValue) {
+      fibonacci = fib1 + fib2;
+      fib1 = fib2;
+      fib2 = fibonacci;
+      fibArray.push(fibonacci);
+    }
+    return fibArray;
+  }
+  fibMaker();
+  for (i = 0; i < fibArray.length; i++) {
```

Comment:
You could also move this logic in the fibMaker above as well. In the end you don't need a function like fibMaker to call, since you'll be doing all the work in one while loop.